src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -3523,6 +3523,193 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
                 );
         }
 
+        [Fact]
+        public void PassingParameters_04()
+        {
+            var source =
+@"interface I<T> { }
+class C
+{
+    static void F(I<object> x, I<object?> y, I<object>? z)
+    {
+        G(x);
+        G(y);
+        G(x, x, x);
+        G(x, y, y);
+        G(x, x, y, z);
+        G(y: x, x: y);
+        G(y: y, x: x);
+        I<object?>[]? w = null;
+        G(x, w);
+        G(x, new I<object?>[0]);
+        G(x, new[] { x, x });
+        G(x, new[] { y, y });
+        G(x, new[] { x, y, z });
+        G(y: new[] { x, x }, x: y);
+        G(y: new[] { y, y }, x: x);
+    }
+    static void G(I<object> x, params I<object?>[] y)
+    {
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (7,11): warning CS8620: Nullability of reference types in argument of type 'I<object?>' doesn't match target type 'I<object>' for parameter 'x' in 'void C.G(I<object> x, params I<object?>[] y)'.
+                //         G(y);
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "y").WithArguments("I<object?>", "I<object>", "x", "void C.G(I<object> x, params I<object?>[] y)").WithLocation(7, 11),
```

Comment:
Hum, it's difficult to track which diagnostic corresponds to which line. Can we add comments (`y /* error 1 */`) or use using local names (`y1`, etc) to make it clearer?